src/pybind/cephfs.py

```diff
@@ -299,6 +307,41 @@ def chdir(self, path):
         if ret < 0:
             raise make_ex(ret, "chdir failed")
 
+    def isdir(self, dirent):
+        return dirent['d_type'] == 0x4
+
+    def opendir(self, path):
+        self.require_state("mounted")
+        if not isinstance(path, str):
+            raise TypeError('path must be a string')
+        dir_handler = c_void_p()
+        ret = self.libcephfs.ceph_opendir(self.cluster, c_char_p(path),
+                                          pointer(dir_handler));
+        if ret < 0:
+            raise make_ex(ret, "opendir failed")
+        return dir_handler
+
+    def readdir(self, dir_handler):
+        self.require_state("mounted")
+        self.libcephfs.ceph_readdir.restype = POINTER(cephfs_dirent)
+        while True:
+            dirent = self.libcephfs.ceph_readdir(self.cluster, dir_handler)
+            if not dirent:
+                return None
+
+            if dirent.contents.d_name != '.' and dirent.contents.d_name != '..':
+                return {'d_ino': dirent.contents.d_ino,
+                        'd_off': dirent.contents.d_off,
+                        'd_reclen': dirent.contents.d_reclen,
+                        'd_type': dirent.contents.d_type,
+                        'd_name': dirent.contents.d_name}
```

Comment:
I'd prefer we leave the '.' and '..' here so that readdir() is equivalent to the C function of the same name, rather than imitating pythons os.listdir